### PR TITLE
Rust MUSL support

### PR DESCRIPTION
**Pull request by slinkydeveloper**
### Checklist:

- [x] Read the [Code of Conduct](https://github.com/appsody/website/blob/master/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md).

- [x] Followed the [commit message guidelines](https://github.com/appsody/website/blob/master/CONTRIBUTING.md#commit-message-guidelines).

- [x] Stack adheres to [Appsody stack structure](https://github.com/appsody/website/blob/master/content/docs/stacks/stack-structure.md).

### Modifying an existing stack:

- [x] Updated the stack version in `stack.yaml`

This PR switches the Rust target to MUSL, allowing static linking of all built binaries. This reduces the production image size (tried with hello world provided by appsody init) from:

```
rust-appsody-test   latest              527bcec585bd        About a minute ago   1.02GB
```

To:

```
rust-appsody-test         latest              9ad501bba26b        26 seconds ago      2.85MB
```

### Related Issues:

Related to https://github.com/appsody/stacks/issues/515

### CODE REVIEW

**Comment by skoh7645** [comment]
Can you also change the path in the `launch.json` config since this PR changes the directory where the binary is generated :)

**Comment by EnriqueL8** [comment]
@slinkydeveloper This PR looks really good, thanks for raising it!

**Comment by EnriqueL8** [comment]
@skoh7645 Could we also change the rocket stack to this? Or does it need the `rust-slim` image to run?

File: experimental/rust/image/Dockerfile-stack [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
-ENV APPSODY_MOUNTS=.:/project/user-app
+ENV APPSODY_MOUNTS=.:/project/user-app;~/.cache/sccache:/project/sccache
 ENV APPSODY_DEPS=$CARGO_HOME
 
 ENV APPSODY_WATCH_DIR="/project/user-app"
 ENV APPSODY_WATCH_REGEX="^(Cargo.toml|.*.rs)$"
 
-ENV APPSODY_RUN="cargo run"
+ENV APPSODY_RUN="cargo run --target x86_64-unknown-linux-musl"
 ENV APPSODY_RUN_ON_CHANGE="cargo run"
```
Comment (skoh7645) [inline experimental/rust/image/Dockerfile-stack]: You need `cargo run --target x86_64-unknown-linux-musl` here too

Comment (slinkydeveloper) [reply experimental/rust/image/Dockerfile-stack]: uops

**Comment by skoh7645** [comment]
@slinkydeveloper Can you also add a user to the production dockerfile so we don't run as root

**Comment by slinkydeveloper** [comment]
@skoh7645 this one https://github.com/appsody/stacks/pull/526/commits/9a39c02fc42020decf8a8691d633cd245a10b415 should do the trick

**Comment by slinkydeveloper** [comment]
I can't manage to reproduce the CI failure locally, any ideas?

**Comment by skoh7645** [comment]
I ran this build on my branch locally with `appsody stack validate` on verbose and found that this was the error: 
Screenshot 2020-01-09 at 13 36 04
So it's not a problem with your stack. There is an issue open on this https://github.com/docker/for-linux/issues/484

**Pull request merged**